src/Razor/src/Microsoft.VisualStudio.RazorExtension/Snippets/SnippetService.cs

```diff
@@ -205,6 +201,25 @@ private ImmutableArray<SnippetInfo> ExtractSnippetInfo(SnippetLanguage language,
         }
     }
 
+    private static ImmutableHashSet<string> GetIgnoredSnippets(SnippetLanguage language, SnippetSetting snippetSetting)
+    {
+        if (language == SnippetLanguage.CSharp)
+        {
+            // In call cases for C# we want to filter out the built in snippets. The C#
+            // language server will handle returning these and isn't required for the
+            // Razor code to add them.
+            return s_builtInSnippets[s_CSharpLanguageId];
```

Comment:
Love this, much clearer, however should we:

a) Have a note on the setting that it only applies to html?
b) Remove (or turn off) the handling of C# snippets from Roslyn, since we can do it now?

Actually, I just realised snippets are still disabled for C#, right? So we don't need either until that changes.

Reply:
Wait.. maybe the fact that C# is off is _more_ reason to add a note to the setting, lest people complain that they can't get rid of `propdp` 😛

Reply:
I'm going to leave for now just because I think wording is already kind of hard to follow. Preview 1 lets see if anyone complains. We can also think about adding this setting to how we request completion to roslyn as an extra bool to add snippets or not. 

Basically, I consider those bugs as _known_ bugs and don't want to confuse people too much. Instead I'd rather them think it is exactly that, a bug.